### PR TITLE
Actions - Google Tag Manager

### LINKED ISSUE #6841: [ACTION] Google Tag Manager
**Is there a specific app this action is for?**
Google Tag Manager

**Please provide a link to the relevant API docs for the specific service / operation.**
https://developers.google.com/tag-platform/tag-manager/api/v2/reference/accounts/containers/workspaces/tags#resource

**Actions:**
1. List GTM Tags (https://developers.google.com/tag-platform/tag-manager/api/v2/reference/accounts/containers/workspaces/tags/list)
2. Create GTM Tag (https://developers.google.com/tag-platform/tag-manager/api/v2/reference/accounts/containers/workspaces/tags/create)
3. Get GTM Tag (https://developers.google.com/tag-platform/tag-manager/api/v2/reference/accounts/containers/workspaces/tags/get)
4. Update GTM Tag (https://developers.google.com/tag-platform/tag-manager/api/v2/reference/accounts/containers/workspaces/tags/update)

### PR DESCRIPTION
**Pull request by vellames**
Closes #6841 

### <samp>🤖 Generated by Copilot at be61baa</samp>

This pull request adds a new component for Google Tag Manager that allows users to create, get, and update tags in a workspace. The component consists of an app file that defines the common props and the API methods, and four actions that implement the tag operations. The pull request also deletes some unused files, adds a new helper file for the tag priority types, and updates the package.json and pnpm-lock.yaml files.

### <samp>🤖 Generated by Copilot at be61baa</samp>

> _`google_tag_manager`_
> _New actions and app logic_
> _Winter of coding_

### <samp>🤖 Generated by Copilot at be61baa</samp>

*  Add app logic for Google Tag Manager component ([link](https://github.com/PipedreamHQ/pipedream/pull/6970/files?diff=unified&w=0#diff-c32ef95d71b8854bc90d0c8a02354371d2ba6fdff4aa66f8cc7d57d070df7cdeR1-R198))
* Add actions for creating, getting, and updating tags in a workspace ([link](https://github.com/PipedreamHQ/pipedream/pull/6970/files?diff=unified&w=0#diff-880ab6cd72f1408a09bcd83ac16a63d11be317f0cc0bc697de15a9e325f8cec5R1-R116), [link](https://github.com/PipedreamHQ/pipedream/pull/6970/files?diff=unified&w=0#diff-454597e2c43da43590e0641ad85e2a8b928350d23250d86c18633632405f2017R1-R67), [link](https://github.com/PipedreamHQ/pipedream/pull/6970/files?diff=unified&w=0#diff-7dedef36c86616ef9a93c52d97d3ed4fbe90c0f056d4fbc5342d6afa82ba60b9R1-R54), [link](https://github.com/PipedreamHQ/pipedream/pull/6970/files?diff=unified&w=0#diff-fb008470a72036620c521df700f24e5034106b263954a36067ade22b1a1f49cbR1-R128))
* Add helper file for exporting priority types for tag parameters ([link](https://github.com/PipedreamHQ/pipedream/pull/6970/files?diff=unified&w=0#diff-d7ea37f988d68b0e018c8be5c9dc84eeffbae3e73a81f3d147c9670cd55d92b3R1-R32))
* Update package.json file to add `@pipedream/platform` dependency and bump version to 0.0.2 ([link](https://github.com/PipedreamHQ/pipedream/pull/6970/files?diff=unified&w=0#diff-a0c0ed9221b3c6ef2ee153221491bbc3fe5bbf2e077776f7f43b5a46f3f64534L3-R17))
* Update pnpm-lock.yaml file to reflect new dependency and version of the package ([link](https://github.com/PipedreamHQ/pipedream/pull/6970/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL1907-R1910))
* Delete unused files ([link](https://github.com/PipedreamHQ/pipedream/pull/6970/files?diff=unified&w=0#diff-525e1697e43f1bf634b90a190ebf7bf6a7aedd0c3c99440ebc5643fd411c1b2b), [link](https://github.com/PipedreamHQ/pipedream/pull/6970/files?diff=unified&w=0#diff-fb124aceda5b8c8e0555243fe24bb03e8e0d06ed5432c941a96626648003122a))

### CODE REVIEW

File: components/google_tag_manager/actions/create-tag/create-tag.mjs [hunk 1]
```diff
@@ -0,0 +1,116 @@
+import app from "../../google_tag_manager.app.mjs";
+
+export default {
+  name: "Create Tag",
+  version: "0.0.1",
+  key: "google_tag_manager-create-tag",
+  description: "Create a tag in a workspace. [See documentation here](https://developers.google.com/tag-platform/tag-manager/api/v2/reference/accounts/containers/workspaces/tags/create)",
```
Comment (alysonturing) [inline components/google_tag_manager/actions/create-tag/create-tag.mjs:7]:
```suggestion
  description: "Create a tag in a workspace. [See the documentation](https://developers.google.com/tag-platform/tag-manager/api/v2/reference/accounts/containers/workspaces/tags/create)",
```

File: components/google_tag_manager/actions/create-tag/create-tag.mjs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    },
+    monitoringMetadata: {
+      propDefinition: [
+        app,
+        "monitoringMetadata",
+      ],
+    },
+  },
+  async run({ $ }) {
+    const parameter = typeof this.parameter === "string"
```
Comment (alysonturing) [inline components/google_tag_manager/actions/create-tag/create-tag.mjs:84]: you could move this logic to a method, and return the `JSON.parse` or itself accordingly, for better reusability.

File: components/google_tag_manager/actions/get-tag/get-tag.mjs [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import app from "../../google_tag_manager.app.mjs";
+
+export default {
+  name: "Get Tag",
+  version: "0.0.2",
```
Comment (alysonturing) [inline components/google_tag_manager/actions/get-tag/get-tag.mjs:5]:
```suggestion
  version: "0.0.1",
```

File: components/google_tag_manager/actions/get-tag/get-tag.mjs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  key: "google_tag_manager-get-tag",
+  description: "Get a specific tag of a workspace. [See documentation here](https://developers.google.com/tag-platform/tag-manager/api/v2/reference/accounts/containers/workspaces/tags/get)",
```
Comment (alysonturing) [inline components/google_tag_manager/actions/get-tag/get-tag.mjs:7]:
```suggestion
  description: "Get a specific tag of a workspace. [See the documentation](https://developers.google.com/tag-platform/tag-manager/api/v2/reference/accounts/containers/workspaces/tags/get)",
```

File: components/google_tag_manager/actions/get-tags/get-tags.mjs [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import app from "../../google_tag_manager.app.mjs";
+
+export default {
+  name: "Get Tags",
+  version: "0.0.1",
+  key: "google_tag_manager-get-tags",
+  description: "Get all tags of an workspace. [See documentation here](https://developers.google.com/tag-platform/tag-manager/api/v2/reference/accounts/containers/workspaces/tags/list)",
```
Comment (alysonturing) [inline components/google_tag_manager/actions/get-tags/get-tags.mjs:7]:
```suggestion
  description: "Get all tags of an workspace. [See the documentation](https://developers.google.com/tag-platform/tag-manager/api/v2/reference/accounts/containers/workspaces/tags/list)",
```

File: components/google_tag_manager/actions/update-tag/update-tag.mjs [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    },
+    monitoringMetadata: {
+      propDefinition: [
+        app,
+        "monitoringMetadata",
+      ],
+    },
+  },
+  async run({ $ }) {
+    const parameter = typeof this.parameter === "string"
```
Comment (alysonturing) [inline components/google_tag_manager/actions/update-tag/update-tag.mjs:96]: you could move this logic to a method, and return the `JSON.parse` or itself accordingly, for better reusability.

**Review by alysonturing — COMMENTED** [review]
LGTM, I'm moving it forward, but there's are some minor issues, please fix it before merge.

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[Google_tag_6841_1118.pdf](https://github.com/PipedreamHQ/pipedream/files/11828570/Google_tag_6841_1118.pdf)

**Review by alysonturing — APPROVED** [review]

File: components/google_tag_manager/actions/update-tag/update-tag.mjs [hunk 7]
```diff
@@ -2,7 +2,7 @@ import app from "../../google_tag_manager.app.mjs";
 
 export default {
   name: "Update Tag",
-  version: "0.0.1",
+  version: "0.0.1687406578",
```
[the remark this answers is not in the document]
Comment (vellames) [reply components/google_tag_manager/actions/update-tag/update-tag.mjs]: It is outdated. It is 0.0.1 now

**Pull request merged**